src/main/main.c

```diff
@@ -234,7 +234,7 @@ void init(void)
     servoMixerInit(masterConfig.customServoMixer);
 #endif
 
-#ifdef SPEKTRUM_BIND
+#if defined(SERIAL_RX) && defined(SPEKTRUM_BIND)
```

Comment:
Maybe separate 'config sanitization' .h file included just after `config.h` may solve this - locate  defaults here (`#ifndef X  #define X <default>`, sanitize values from config.h (possibly with warning) etc ..